Common/GPU/thin3d.cpp

```diff
@@ -623,6 +623,46 @@ void ConvertToD32F(uint8_t *dst, const uint8_t *src, uint32_t dstStride, uint32_
 	}
 }
 
+void ConvertToD16(uint8_t *dst, const uint8_t *src, uint32_t dstStride, uint32_t srcStride, uint32_t width, uint32_t height, DataFormat format) {
+	if (format == Draw::DataFormat::D32F) {
+		const float *src32 = (const float *)src;
+		uint16_t *dst16 = (uint16_t *)dst;
+		if (src == dst) {
+			return;
+		} else {
+			for (uint32_t y = 0; y < height; ++y) {
+				for (uint32_t x = 0; x < width; ++x) {
+					dst16[x] = (uint16_t)(src32[x] * 65535.0f);
```

Comment:
This won't be a correct conversion for our purposes, though?  I thought I wrote some logic in... I guess it was #11676?

Under some versions of our depth support, we intentionally use 1/4 of the range (in the middle) for the actual depth values, where the rest is meant to be clamped.  This reduces the bugs from not having depth clamp.  If we have depth clamp, I think we use the full range, though.

We need to account for that somewhere (maybe not in thin3d, not sure) for 24 or 32 bit depth buffers.

-[Unknown]

Reply:
Yeah, that's right. I'll leave a TODO for now, since I might indeed try to address this elsewhere. This'll work when we don't reduce the range which is most of the time on Vulkan, for example.